### ISSUE TITLE
q: is nominating from a Ledger account supported?

### ISSUE THREAD

**Issue by dvdplm** [polkadot-js/apps#3774]
I just tried to nominate on Polkadot and got an error at the last step:

```
signAndSend: error: Error: Not supported
    at t.default (polkadot.02.b22b8c2a.js:1)
    at t.default._wrapError (polkadot.02.b22b8c2a.js:1)
    at /apps/async https:/polkadot.js.org/apps/polkadot.02.b22b8c2a.js:1
    at async t.default._withApp (polkadot.02.b22b8c2a.js:1)
    at async t.default.signPayload (main.1bbc4e22.js:374)
    at async Object.value (polkadot.01.d940f4a0.js:1)
    at async e.project (polkadot.01.d940f4a0.js:1)
(anonymous) @ main.1bbc4e22.js:374
```

Is nominating from a ledger not supported yet?

**Comment by jacogr** [comment]
You cannot do „+ Nominate“ or „+ Validate“ from a Ledger. S e the documentation Ledger does not support utility.batch decoding. Until it does use the „+ Stash“. 

(All explained in the wiki, don’t have a timeline of that team having support)

**Comment by dvdplm** [comment]
Is it useful to you if I open a ticket for handling the incompatibility in a more user friendly way? As is, the user experience is a bit rough.

**Comment by jacogr** [comment]
This should be addressed on the Ledger side. So this is what happens 

- UI takes you through a modal, combining steps
- take each of these steps and creates a utility.batch fo single submission (The stash does not do this, it forces you to make a transaction one at a time)
- it send the tx (encoded) for signing (where a signer can be a normal account, extension signer, QR signer or hardware signer)
- In this case the Ledger itself does not have support for the utility.batch construct (nested calls), so it fails. 
- the UI reports the error exactly as received (it doesn't understand this or any other error, just that "something went wrong")

For staking we kept the old "+ stash" (manual step) even when adding the one-click validation/nomination options. There are  some areas such as payouts are only supported via batch, there is no fallback since it does mean multiple calls (up to block limits) if multiple payouts are available.

I honestly don't want to start building in filters for specific types of accounts since cannot tracks the development on all. This means that everywhere there needs to be something that tries to find out the account type and then do magic to not show these - which means support calls are then logged at that point, i.e. "account does not show up".

TL;DR There is complexity in batch calls (theoretically infinitely nested, not a good idea on limited devices), however the basic use-case should be supported on the hardware side. (Like for instance it can do nested calls such as proposals or sudo). Not sure if there is a tracking issue on the Ledger app side, but it is something that the team there is aware of.

**Comment by jacogr** [comment]
Ledger tracking issue is here - https://github.com/Zondax/ledger-polkadot/issues/16

**Comment by dvdplm** [comment]
Thank you for taking the time explaining. As a novice user I always appreciate whenever the UI includes some in-line help. On a normal web app nobody reads anything ever but this one here is a bit different. People use it with a different mindset perhaps. All of that to say, adding a note in the UI about known limitations to the hardware wallets would be good. And of course improve the wiki docs wrt to Ledger (I believe this is underway elsewhere).

**Comment by jacogr** [comment]
Like mentioned, there is limited resourcing here and I certainly will not track what is and what is not supported in the different signers. Like networks and capabilities, it is an impossible task for what is available in the time available.
